elixir/apps/web/lib/web/live/sign_up.ex

```diff
@@ -0,0 +1,234 @@
+    </h3>
+    <.simple_form for={@form} class="space-y-4 lg:space-y-6" phx-submit="submit" phx-change="validate">
+      <.inputs_for :let={account} field={@form[:account]}>
+        <.input
+          field={account[:name]}
+          type="text"
+          label="Account Name"
+          placeholder="Enter an Account Name here"
+          required
+          phx-debounce="300"
+        />
+        <.input field={account[:slug]} type="hidden" />
+      </.inputs_for>
+
+      <.inputs_for :let={actor} field={@form[:actor]}>
+        <.input
+          field={actor[:name]}
+          type="text"
+          label="Your Name"
+          placeholder="Enter your name here"
+          required
+          phx-debounce="300"
+        />
+        <.input field={actor[:type]} type="hidden" />
+      </.inputs_for>
+
+      <.input
+        field={@form[:email]}
+        type="text"
+        label="Email"
+        placeholder="Enter your email here"
+        required
+        phx-debounce="300"
+      />
+
+      <:actions>
+        <.button phx-disable-with="Creating Account..." class="w-full">
+          Create Account
+        </.button>
+      </:actions>
+      <p class="text-xs text-center">
+        By signing up you agree to our <.link
+          href="https://www.firezone.dev/terms"
+          class="text-blue-600 dark:text-blue-500 hover:underline"
+        >Terms of Use</.link>.
+      </p>
+    </.simple_form>
+    """
+  end
+
+  def handle_event("validate", %{"registration" => attrs}, socket) do
+    changeset =
+      %Registration{}
+      |> Web.Registration.changeset(attrs)
+      |> Map.put(:action, :validate)
+
+    socket = assign(socket, form: to_form(changeset))
+
+    {:noreply, socket}
+  end
+
+  def handle_event("submit", %{"registration" => orig_attrs}, socket) do
+    attrs = put_in(orig_attrs, ["actor", "type"], :account_admin_user)
+
+    changeset =
+      %Registration{}
+      |> Web.Registration.changeset(attrs)
+      |> Map.put(:action, :insert)
+
+    if changeset.valid? do
+      registration = Ecto.Changeset.apply_changes(changeset)
+
+      multi =
+        Ecto.Multi.new()
+        |> Ecto.Multi.run(
+          :account,
+          fn _repo, _changes ->
+            Accounts.create_account(%{
+              name: registration.account.name
+            })
+          end
+        )
+        |> Ecto.Multi.run(
+          :provider,
+          fn _repo, %{account: account} ->
+            Auth.create_provider(account, %{
+              name: "Magic Link",
+              adapter: :email,
+              adapter_config: %{}
+            })
+          end
+        )
+        |> Ecto.Multi.run(
+          :actor,
+          fn _repo, %{provider: provider} ->
+            Actors.create_actor(provider, registration.email, %{
+              type: :account_admin_user,
+              name: registration.actor.name
+            })
+          end
+        )
+
+      Domain.Repo.transaction(multi)
+      |> case do
```

Comment:
This feels a bit of an abuse of a pipe operator, we can either remove it here:
```suggestion      
      case Domain.Repo.transaction(multi) do
```

or use all the way with multi:

```
)
|> Domain.Repo.transaction()
|> case do
```

Reply:
👍 Sounds good.  Do you have a preference on the options you described?

Reply:
This has been updated